backend/src/main/scala/cromwell/backend/standard/callcaching/StandardCacheHitCopyingActor.scala

```diff
@@ -238,16 +256,112 @@ abstract class StandardCacheHitCopyingActor(val standardParams: StandardCacheHit
     }
   }
 
-  /* Blacklist by prefix if appropriate. */
-  private def handleForbidden[T](path: String, andThen: => State): State = {
+  /* Blacklist by bucket and hit if appropriate. */
+  private def handleBlacklistingForForbidden[T](path: String, andThen: => State): State = {
     for {
+      // Blacklist the hit first in the forcomp since not all configurations will support bucket blacklisting.
       cache <- standardParams.blacklistCache
+      _ = blacklistAndMetricHit(cache, standardParams.cacheHit)
       prefix <- extractBlacklistPrefix(path)
-      _ = cache.blacklist(prefix)
+      _ = blacklistAndMetricBucket(cache, prefix)
     } yield()
     andThen
   }
 
+  private def handleBlacklistingForGenericFailure(): Unit = {
+    standardParams.blacklistCache foreach { blacklistAndMetricHit(_, standardParams.cacheHit) }
+  }
+
+  /* Whitelist by bucket and hit if appropriate. */
+  private def handleWhitelistingForSuccess(command: IoCommand[_]): Unit = {
+    for {
+      cache <- standardParams.blacklistCache
+      _ = whitelistAndMetricHit(cache, standardParams.cacheHit)
+      copy <- Option(command) collect { case c: IoCopyCommand => c }
+      prefix <- extractBlacklistPrefix(copy.source.toString)
+      _ = whitelistAndMetricBucket(cache, prefix)
+    } yield ()
+    ()
+  }
+
+  private def publishBlacklistMetric(blacklistCache: BlacklistCache, verb: String, bucketOrHit: String, key: String, value: BlacklistStatus): Unit = {
+    // TODO probably remove the logging after interactive testing is complete
+    log.info("Publishing blacklist metric: {} {} {} {}", verb, bucketOrHit, key, value)
```

Comment:
Might be moot since we probably want to delete this, but it would be easier for me to parse as `Group x's Hit blacklist returned KnownGood when queried for key xyz`